### ISSUE TITLE
Plugin causes files within movie .iso images to be scanned into library

### ISSUE THREAD

**Issue by dabl2** [xbmc/audiodecoder.sacd#11]
For users who have movie .iso images, with this plugin enabled, when using 'update Library' for Movies, multiple instances of files within movie .iso images such as CERTIFICATE\id.bdmv are added to the Movie library as bogus titles such as 'Kung Fu Hustle'. These have a udf://../../.. prefix and will be removed when cleaning the library.

Further, movie .iso images that do not have an entry in the themoviedb.org database but do have a local .nfo are not added to the library.

Disabling the plugin 'fixes' both these problems.

Might it be possible to have an option to link/assign the plugin to only 'see' .iso files in certain folders and/or ignore one or more folders (ex where movie .iso images are located)?

See other reports [here](https://forum.kodi.tv/showthread.php?tid=106329&pid=3012772#pid3012772) including ones suggesting it also picks up .id extensions (eg bdmv.id)

**Comment by Reemtsman** [comment]
i can confirmt the problem described above. I had hundreds of "Kung Fu Hustles" entries in my library after the last update. 
I just started a own issue report as i did not realize the same problem is already reported here. Sorry for the duplication!

**Comment by dabl2** [comment]
For what it's worth the good news is I found that simply running a kodi library clean operation gets rid of the bogus movie entries created by the plugin being active when running a scan against a target containing .iso files.

**Comment by Reemtsman** [comment]
@dabl2 dabl2! From my experience the workaround suggested by you only works when the SACD Iso Support Add-on is De-Installed or at least Deactivated an the time of the Cleanup.
Therefore more a Workaround, not a solution. But still thank you for posting!!

**Comment by BenMcLean** [comment]
Suggestion: Maybe re-name all the sacd ISOs to be `filename.iso.sacd` and then have this plugin look for `.sacd` files?

**Comment by Reemtsman** [comment]
Thank you @BenMcLean !
Unfortunately this Plugin has no setting to define what type of extension to look for.

**Comment by dabl2** [comment]
> Thank you @BenMcLean !
> Unfortunately this Plugin has no setting to define what type of extension to look for.

But I think Ben is suggesting that maybe it *could* 

If of course some developer saw fit to consider this idea.

If I understand Ben correctly, that idea does come with a cost though.  

Asking the user to rename all their files which otherwise might then become useless to other apps is perhaps after further thought, maybe not ideal.

I still like the idea of allowing the user to 'aim' the app at one or more folders to pay attention to/act upon.

IE I think most people have their SACD .ISOs in a folder seperate from their Movie .ISOs so this should work fine for most people.

I'd be happy at this point just to see this issue even acknowledged.

**Comment by BenMcLean** [comment]
Yes that was a feature suggestion for the developer, not a workaround.

Personally, I haven't managed to get this plugin to work at all. :(

I install it but it seems to have no effect

**Comment by Reemtsman** [comment]
...sorry for my misunderstanding!
But i agree with Don that the file extension is not the smartest approach to trigger the correction. I would suggest to trigger this through the source path. I would bet that 99% of all users have their SACD Isos not in the same path like the video Isos anyhow.
Just let the user enter his SACD Source Path into the Add-on and make the addon only work for the Isos found under this predefined SACD path.

**Comment by BenMcLean** [comment]
No way. My collection is organized by artist, not by format. Don't do that path crap.

**Comment by Reemtsman** [comment]
...your SACD Iso´s lay under Artists...
So what?? This means they are located in your music library wich is by design a different source (with a different path) than your movies and TV Shows (with Video Isos). So triggering the behaviou of this Add-on through the source patz is definitely smarter than renaming all SACD.Iso´s

**Comment by BenMcLean** [comment]
No, all formats are together in the same artist folder. Like the Jeff Wayne folder has all the Jeff Wayne SACDs and all the Jeff Wayne DVD ISOs.

**Comment by dabl2** [comment]
In theory a developer could implement support for either/both options.

**Comment by Reemtsman** [comment]
The challenhe here will be to make sure this Add-on exclusively effects SACD Iso´s without effecting Video-Iso´s. As lond as this is acchieved - without renaming all my SACD Iso´s - i will be happy.

**Comment by AlwinEsch** [comment]
About the problem becomes here a half fix added https://github.com/xbmc/audiodecoder.sacd/pull/35

Add the option and related code changes so that ISO's can also have ".sacd" as an extension. 

However, I want to leave ".iso" in there because more is correct and available than such. In addition, the error is in Kodi and must be looked at to fix it.

But you can now edit the addon.xml by hand and remove the text with ".iso".
https://github.com/AlwinEsch/audiodecoder.sacd/blob/Matrix-change/audiodecoder.sacd/addon.xml.in#L11

**Comment by BenMcLean** [comment]
Thank you.

> the error is in Kodi and must be looked at to fix it.

Is there a Github issue made for that which I can subscribe to?

**Comment by AlwinEsch** [comment]
Unfortunately none at the moment, as far as I know.

**Comment by BenMcLean** [comment]
> Unfortunately none at the moment, as far as I know.

OK ... can somebody who knows more about the innards of Kodi than I do write it?

I could write it but I'd probably say the wrong thing

**Comment by AlwinEsch** [comment]
Have created here a fix request https://github.com/xbmc/xbmc/pull/20205.

Hope it can be become in on Nexus and also on Matrix version.
